osu.Game/Skinning/PoolableSkinnableSample.cs

```diff
@@ -132,6 +132,8 @@ public void Play()
             if (Sample == null)
                 return;
 
+            FlushPendingSkinChanges();
+
```

Comment:
This is why we write test coverage of changes: because you miss silly things like this otherwise.

This call needs to be before the `null` check above, else it's never going to be hit (`Sample` is nulled when a change occurs, aka the thing you're looking to fix).